opencog/cython/opencog/atomspace_details.pyx

```diff
@@ -89,6 +89,12 @@ cdef class AtomSpace:
             atom = self.add_link(t, out, tv)
         return atom
 
+    def add_atom(self, Atom atom):
+        cdef cHandle result = self.atomspace.add_atom(atom.get_c_handle())
+        if result == result.UNDEFINED:
```

Comment:
Rather than returning nullptr, the `AtomTable` code should almost surely be changed to throw an exception. That would allow you to skip this check.

Reply:
What if atom is DeleteLink?

Reply:
Oh. Interesting. I'll have to think about that. It still might be the right thing to throw ... 

Not all exceptions in the atomspace are "errors". Many are simply "fast paths" that are used to indicate "unusual conditions" (e.g. the `SilentException`) thus avoiding a useless if-statement for 99.9% of the cases, and paying the exception overhead for the 0.1% of the cases that need it.  It seems that this might  be the case here, too, with most cases being "actual errors", while the `DeleteLink` maybe should be a `SilentException`.

This is not urgent, its just .. a general goal to move towards.